.github/workflows/post-release-mergeback.yml

```
@@ -0,0 +1,104 @@
+# This workflow runs after a release of the action.
+# It merges any changes from the release back into the
+# main branch. Typically, this is just a single commit
+# that updates the changelog.
+name: Mergeback after release
+
+on:
+  workflow_dispatch:
+    inputs:
+      baseBranch:
+        description: 'The base branch to merge into'
+        default: main
+        required: false
+
+  # Commented for now to ensure we don't run this accidentally after a real merge
+  # push:
+  #   branches:
+  #     - v1
+
+  pull_request:
+    paths:
+      - .github/workflows/post-release-mergeback.yml
+
+jobs:
+  merge-back:
+    runs-on: ubuntu-latest
+    env:
+      BASE_BRANCH: "${{ github.event.inputs.baseBranch || 'main' }}"
+      # TODO head_ref won't exist in workflow dispatch, but not sure what to put here until I can start trying this out using workflow_dispatch
+      HEAD_BRANCH: "${{ github.head_ref }}"
+      NEW_BRANCH: "mergeback/${{ github.head_ref }}-to-${{ github.event.inputs.baseBranch || 'main' }}"
+      DEFAULT_REVIEWER: "github/codeql-core"
+
+    steps:
+      - name: Dump GitHub context
+        env:
+          GITHUB_CONTEXT: '${{ toJson(github) }}'
+        run: echo "$GITHUB_CONTEXT"
+
+      - name: Dump branches
+        run: |
+          echo "BASE_BRANCH $BASE_BRANCH"
+          echo "HEAD_BRANCH $HEAD_BRANCH"
+          echo "NEW_BRANCH $NEW_BRANCH"
+
+      - name: validate
+        run: |
+          if [ "$BASE_BRANCH" -eq "$HEAD_BRANCH" ]; then
+            echo "error::Base branch and head branch are the same $BASE_BRANCH."
+            exit -1
+          fi
+
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+
+      - name: Update git config
+        run: |
+          git config --global user.email "codeql-core@github.com"
+          git config --global user.name "CodeQL Actions Bot"
```

This will make it look like the global Actions bot user.
```suggestion
          git config --global user.email "github-actions@github.com"
          git config --global user.name "github-actions[bot]"
```